src/services/ui/src/utils/DataPoller.ts

```diff
@@ -0,0 +1,46 @@
+export type Options<TFetcherReturn> = {
+  fetcher: () => TFetcherReturn;
+  checkStatus: (data: Awaited<TFetcherReturn>) => boolean;
+  interval: number;
+  pollAttempts: number;
+};
+
+export class DataPoller<TFetcherReturn> {
```

Comment:
i think we can build poller in here and only export that:

```javascript
export const syncStatusPoller = new DataPoller({
    interval: 1000,
    pollAttempts: 20,
    fetcher: () => getItem(formData.id),
    checkStatus: (data) => {
        return data._source.actionType === "Renew"; // map this instead of hard-code
    },
});
```

and elsewhere just use it by

```javascript
import { syncStatusPoller } from "..."
await syncStatusPoller.startPollingData();
```

Reply:
I like this idea a lot. I definitely will look into adding that

Reply:
yeah! the idea is we could have different intervals and attempts and what not if we need other pollers

Reply:
I'm going to generalize the name of checkStatus cause this is not a concept the poller concerns itself with